### DISCUSSION TITLE
Helix on Remote Machine (AWS EC2) is laggy when connected via browser

### DISCUSSION THREAD

**Discussion by mert-kurttutan** [helix-editor/helix#10846] [Troubleshooting]
I am testing out helix on a remote machine in aws (connected with in-browser option)
Whenever I move out of the current (visible) part of file, helix is very laggy and slow (i.e. it takes longer for it to move to the next line).
This problem does not occur on nano or neovim.

Any insight into this. Is this expected?

**Comment by dead10ck** [comment]
This could be any number of things. Network latency, how helix was built / installed, the size of the file and how optimized its grammar is, etc. It's likely this has either everything or nothing to do with being remote. There's not really enough information here to help diagnose.

**Comment by mert-kurttutan** [comment]
I don't think my network is an issue (at least not an issue for all the other things I do remotely, e.g. use of other editor with remote, colab notebook etc), I also checked the download/upload speed. 

As I said, this happens only with helix. I tested it the same file with 3 different editor, neovim, vim and nano. This happens only with helix.

Further info:
I reproduced the issue with the following machine.
Aws ec2 t2.micro
Os: Ubuntu 24

My steps:
- Launch the machine above
- Use aws inbrowser option to connect to the machine
- Install helix 
- Open any file with some sufficient text (For reference you can use https://raw.githubusercontent.com/itsfoss/text-files/master/agatha_complete.txt).
- Move through the file with up/down arrows and compare it to the other editors.

**Reply by markstos** [reply]
Could you create a screen-share video that illustrates the difference between helix and another editor?

**Reply by dead10ck** [reply]
Could you expand on "install helix"? How did you install it?

**Reply by mert-kurttutan** [reply]
> Could you expand on "install helix"? How did you install it?

Using instructions given in the ubuntu section of https://docs.helix-editor.com/install.html

**Reply by dead10ck** [reply]
Ok, I also would like to see a screenshare of the problem. 

Also, did you try connecting to your ec2 instance from a real terminal, rather than your web browser? i.e. with the AWS CLI with a SSM session, or ssh?

**Reply by mert-kurttutan** [reply]
> Ok, I also would like to see a screenshare of the problem.
> 
> Also, did you try connecting to your ec2 instance from a real terminal, rather than your web browser? i.e. with the AWS CLI with a SSM session, or ssh?

I just tried it with ssh from my terminal. Now, there is a lagging issue.

**Reply by mert-kurttutan** [reply]
Here is a video of the lagging when connected via browser (compared with neovim)
[Screencast from 29.05.2024 19_13_30.webm](https://github.com/helix-editor/helix/assets/88637659/d2ab4f2e-d617-40bb-9294-a5a9cd220472)

**Reply by markstos** [reply]
The video does a great job of illustrating the issue. If you SSH in a second window and run htop/top during the experiment, you can you see a difference in the CPU usage of the two? (And whether that CPU difference is from the helix binary or a language server?).

**Reply by mert-kurttutan** [reply]
> The video does a great job of illustrating the issue. If you SSH in a second window and run htop/top during the experiment, you can you see a difference in the CPU usage of the two? (And whether that CPU difference is from the helix binary or a language server?).

I can try this tomorrow. I am not sure about LSP side of things, but I observed the issue with txt file as well.

**Comment by gabydd** [comment]
Do your files have very long lines or lots of softwrapped lines that can cause a slow down at the moment especially if the computer is not very powerful?

**Reply by mert-kurttutan** [reply]
you check the reference file I put in my 2nd message. To your question, no they dont (including the reference file).
